client/remote_test.go

```diff
@@ -39,85 +40,59 @@ var (
 )
 
 // Set up compatible server and client for use by tests.
-func TestMain(t *testing.T) {
-	var err error
+func TestMain(m *testing.M) {
 	var pemBytes []byte
 	var p *pem.Block
 	var priv crypto.Signer
 
 	// Setup keyless server
-	s, err = server.NewServerFromFile(serverCert, serverKey, keylessCA, serverAddr, socketAddr)
-	if err != nil {
-		t.Fatal(err)
-	}
+	s, _ = server.NewServerFromFile(serverCert, serverKey, keylessCA, serverAddr, socketAddr)
 
 	keys := server.NewDefaultKeystore()
 	s.Keys = keys
-	if pemBytes, err = ioutil.ReadFile(rsaPrivKey); err != nil {
-		t.Fatal(err)
-	}
+	pemBytes, _ = ioutil.ReadFile(rsaPrivKey)
 	p, _ = pem.Decode(pemBytes)
-	if priv, err = x509.ParsePKCS1PrivateKey(p.Bytes); err != nil {
-		t.Fatal(err)
-	}
-	if err = keys.Add(nil, priv); err != nil {
-		t.Fatal(err)
-	}
+	priv, _ = x509.ParsePKCS1PrivateKey(p.Bytes)
 
-	if pemBytes, err = ioutil.ReadFile(ecdsaPrivKey); err != nil {
-		t.Fatal(err)
-	}
+	keys.Add(nil, priv)
+
+	pemBytes, _ = ioutil.ReadFile(ecdsaPrivKey)
 	p, _ = pem.Decode(pemBytes)
-	if priv, err = x509.ParseECPrivateKey(p.Bytes); err != nil {
-		t.Fatal(err)
-	}
-	if err = keys.Add(nil, priv); err != nil {
-		t.Fatal(err)
-	}
+	priv, _ = x509.ParseECPrivateKey(p.Bytes)
+	keys.Add(nil, priv)
 
 	go func() {
-		if err := s.ListenAndServe(); err != nil {
-			t.Fatal(err)
-		}
+		s.ListenAndServe()
 	}()
 
 	go func() {
-		if err := s.UnixListenAndServe(); err != nil {
-			t.Fatal(err)
-		}
+		s.UnixListenAndServe()
 	}()
 
 	// wait for server to start
 	time.Sleep(100 * time.Millisecond)
 	// Setup keyless client
-	if c, err = NewClientFromFile(clientCert, clientKey, keyserverCA); err != nil {
-		t.Fatal(err)
-	}
+	c, _ = NewClientFromFile(clientCert, clientKey, keyserverCA)
```

Comment:
I haven't figured out how to properly bail out setup errors in TestMain. Or maybe I am using it wrong.